src/internal/mod.rs

```diff
@@ -345,6 +345,71 @@ impl DeviceContext {
     }
 }
 
+impl From<DeviceAddResult> for DeviceContext {
+    fn from(dar: DeviceAddResult) -> Self {
+        DeviceContext::new(
+            dar.account_id,
+            dar.segment_id,
+            dar.device_private_key,
+            dar.signing_private_key,
+        )
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
We need `PartialEq` here. Maybe we should add equality unit tests for all *Result types to ensure that we are providing equality.

I'm also wondering why we are providing `Serialize` and `Deseriaize`. Do we normally provide that? I made a comment on https://github.com/IronCoreLabs/ironoxide/issues/58#issuecomment-574294352 about potentially adding those to all *Result types,  but unless there's a reason for these here, we should probably remove them.

Reply:
We can't add `PartialEq` because we don't have it on the device's `PrivateKey`. I've added a comment explaining, and removed the `Serialize` and `Deserialize`